packages/nimble-components/src/table/specs/action-menu-hld.md

```diff
@@ -0,0 +1,113 @@
+# Nimble Table Action Menu API
+
+## Problem Statement
+
+In some tables, such as many tables within SLE, there is a need to put a menu button within a row of the table. The `nimble-table` should provide a way for clients to accomplish this to avoid code duplication and promote consistency.
+
+### Out of scope of this HLD
+
+-   The interaction of the menu opening and closing in coordination with row selection is out of scope for this HLD. This interaction will be discussed when row selection is defined.
+
+## Links To Relevant Work Items and Reference Material
+
+[Table Spec](./README.md)
+
+Example action menu in SLE
+![SLE action menu](./spec-images/sleActionMenu.png)
+
+## Implementation / Design
+
+### Client-facing API
+
+There are two pieces of configuration that must be provided by a client in order to use the action menu:
+
+1. Specifying which column, or columns, the action menu will be visible within
+2. Specifying the menu and its items to show when the menu button is open
+
+The nimble design system will be opinionated about many details of the menu button within a table. Therefore, a client will not be able to configure the exact details of the menu button itself, such as:
+
+-   The appearance mode of the button
+-   The icon shown within the button
+-   When the button becomes visible (e.g. always visible, visible on hover/selection only, etc.)
+
+If the need arises for a client to have more control of the menu button's configuration, the API can be extended at a later time.
```

Comment:
Yes, the label of the menu button is something we need to provide a way of setting. It needs to be set all the time, not just when the menu is open, so I think that configuration would need to go on the column. I'm thinking something like

```HTML
<nimble-table-column-text value-key="myStr" show-menu menu-label="Item configuration">Column title</nimble-table-column-text>
```

Does that seem reasonable? Or, do you think we should implement it in a way that better forces a client to set the label? For example, we could combine the `show-menu` and `menu-label` attributes into one string attribute, and the menu is shown if that attribute has a non-empty value:
```HTML
<nimble-table-column-text value-key="myStr" menu="Item configuration">Column title</nimble-table-column-text>
```

I personally like the explicit nature of the first option because I think the API is much cleaner.

Reply:
It's tempting to force clients to provide a label but I agree that having separate attributes makes for a more explicit and readable API. I'd also vote for your first option and just encourage setting the `menu-label` in our documentation.

Reply:
Updated.